src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/ServiceConnectionsController.php

```diff
@@ -27,25 +27,50 @@
 use Surfnet\ServiceProviderDashboard\Infrastructure\DashboardBundle\Service\AuthorizationService;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\ExpressionLanguage\Expression;
+use Symfony\Component\Form\FormFactoryInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Attribute\Route;
 use Symfony\Component\Security\Http\Attribute\IsGranted;
 
 class ServiceConnectionsController extends AbstractController
 {
     public function __construct(
-        private readonly AuthorizationService $authorizationService,
+        private readonly AuthorizationService     $authorizationService,
         private readonly ServiceConnectionService $serviceConnectionService,
         private readonly ServiceService $serviceService,
     ) {
     }
 
-    #[IsGranted(new Expression('is_granted("ROLE_ADMINISTRATOR") or is_granted("ROLE_SURFCONEXT_RESPONSIBLE")'))]
+    #[IsGranted(new Expression('is_granted("ROLE_ADMINISTRATOR") or is_granted("ROLE_SURFCONEXT_REPRESENTATIVE")'))]
     #[Route(
         path: '/connections',
         name: 'service_connections',
         methods: ['GET', 'POST']
     )]
-    public function myServices()
+    public function __invoke(): RedirectResponse|Response
+    {
+        // The admin overview (which span multiple services) is rendered in a separate action
+        if ($this->isGranted('ROLE_ADMINISTRATOR')) {
+            return $this->redirect($this->generateUrl('service_admin_overview'));
+        }
+        $entities = $this->serviceConnectionService->findByInstitutionId($this->getInstitutionId());
+        return $this->render(
+            '@Dashboard/Service/my-services.html.twig',
+            [
+                'testIdps' => $entities->getTestIdps(),
+                'entities' => $entities,
+            ]
+        );
+    }
+
+    #[IsGranted("ROLE_ADMINISTRATOR")]
+    #[Route(
```

Comment:
Owww... when I see __invoke, I always assume there's only one route in the controller. I even wrote a rector rule that if a controller has __invoke, it only can have one public method (__invoke)

Consider splitting into two seperate controllers

Reply:
I knew you would come up with this feedback. And you are right of course! I also think that this use case is quite valid (two views on the same data, with a different response format). It seems fitting to have 1 controller with these exact dependencies to serve both purposes.